src/components/ErrorBoundary.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable react/sort-comp */
+import React from 'react';
+import ErrorMessage from './ErrorMessage';
+
+class ErrorBoundary extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { hasError: false };
+  }
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch() {
+    this.props.callTessen();
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <ErrorMessage />;
+    }
+
+    return this.props.children;
+  }
+}
+
```

Comment:
rather than have each call site pass a callback that calls Tessen, i think it might be easier if we export a wrapper component that passes in Tessen so the class component can call it itself. something like

```jsx
const ErrorBoundaryFC = ({ children, eventName }) => {
  const tessen = useTessen()

  return <ErrorBoundary eventName={eventName} tessen={tessen}>{children}</ErrorBoundary>
}

export default ErrorBoundaryFC

// and then in the class component

 componentDidCatch() {
    this.props.tessen.track({
      category: 'PageErrored',
      eventName: this.props.eventName
    });
  }

```

then when we use it we'd only have to do
```jsx
<ErrorBoundary eventName="attributeDictionary">...</ErrorBoundary>
```

Reply:
ooh yeah this is nicer